### PR TITLE
Update locality and localadmin records

### PR DESCRIPTION
**Pull request by stepps00**
Fix https://github.com/whosonfirst-data/whosonfirst-data/issues/538.

This PR will:

- [x] Update matching existing localadmin geometries in the US with 2019 US Census shapes
- [x] Add concordances, source properties, and new label centroids for localadmin
- [x] Add cessation dates / superseding to existing WOF localadmin that are no longer current
- [x] Add new localadmin from the US Census source that are not in WOF
- [x] Update some Township names for records that have single number names (like "6", "9", "D"). Some examples:
  - https://spelunker.whosonfirst.org/id/404499833/
  - https://spelunker.whosonfirst.org/id/404519491/
  - https://spelunker.whosonfirst.org/id/404496791/
- [x] Flag localadmin records in Indiana and Missouri as `mz:hierarchy_label`  = 0
- [x] Update matching existing localities geometries in the US with 2019 US Census shapes
- [x] Add concordances, source properties, and new label centroids for localities
- [x] Add cessation dates / superseding to existing WOF localities that are no longer current
- [x] Add new localities from the US Census source that are not in WOF
- [x] For localadmin that have no descendant locality polygons:
  - Create a new locality record, without the placetype in the `wof:name` or name: properties
  - Flag the new locality and existing localadmin as coterminous
  - Flag the new locality as `wof:statistical_gore` = 1 (or similar way to designate the place is a fill-in feature)
- [x] For localadmin that have one or more descendant locality polygons:
  - Create a new locality record (or update existing locality record if one exists in WOF already), without the placetype in the `wof:name` or `name:` properties
  - The new locality polygon should be the parent localadmin polygon, minus any existing locality polygons in the same localadmin
  - Flag the new/updated locality as `wof:statistical_gore` = 1 (or similar way to designate the place is a fill-in feature)
  - Do not flag the new/updated locality and existing localadmin as coterminous
- [x] QA `mz:hierarchy_label` properties for localadmin in US
- [x] Validate repo/properties
- [x] PIP
- [x] QA changes. Specifically, make sure some manually set hierarchies (like the NYC locality) were not reset.
- [x] Update `min_zoom` on "fill-in" localities (see comment below).
- [x] Update the `whosonfirst-properties` repo with a better description for `wof:coterminous` and `wof:hierarchy_label`. When both are set on a record, special considerations should be made so users are not appending extra information to an address string.
- [x] Ensure all records are indexed (`marketareas`, too)

### CODE REVIEW

**Comment by stepps00** [comment]
**Commit Notes:**
2910289: Updates matched _localadmin_  wof <> uscensus records. Stores existing geometries as alts, adds top-level properties, updates centroids.
672f178: Creates new localadmin records. In some cases, includes the superseding <> superseded_by links and flagging outdated localadmin as not current. Non-current localadmin in this commit were given a cessation date of "2019%" to signify "uncertain andn approximate" ([edtf spec](https://www.loc.gov/standards/datetime/#:~:text=The%20Extended%20Date%2FTime%20Format,a%20wide%20variety%20of%20applications.))..
440d789: Exports localadmin and ensures localadmin in IN and MO have is_hierarchy_label = 0.
0837400: Updates matched _locality_ wof <> uscensus records. Stores existing geometries as alts, adds top-level properties, updates centroids.

**Comment by stepps00** [comment]
@nvkelso - this PR is ready for review. Once approved I will PIP this PR to update hierarchies and complete any remaining work (validating, properties repo update, etc).

Just one issue with the " Update some Township names for records that have single number names (like "6", "9", "D")." item above. In most of these cases, these localadmin names come directly from the US Census. The actual names of a handful of localadmin or simply "06" or "9" (not "Township 06" or "Township 9" as I first suspected).

Do you have any suggestions for what to do with locality and localadmin records in these cases?

**Comment by nvkelso** [comment]
Please also set a `mz:min_zoom` of **30** on the fake localities so they don't get map labels.

**Comment by stepps00** [comment]
PIP work is ongoing for this PR. Once complete, I'll wrap up the smaller remaining tasks and merge.

**Comment by nvkelso** [comment]
@stepps00 Can you provide an update on this one, please?

**Comment by stepps00** [comment]
This PR is still being processed against the PIP tools to update hierarchies for all localadmin and descendant records.

I kicked off a first pass at the PIP work a few weeks ago, which finished but resulted in some buggy hierarchies due to a discrepancy in the branch and what I indexed in the PIP tools. I kicked off a second pass over a week ago which is still processing.

I think the lengthy processing time is due to the amount of files needing to be PIP'd, as well as the size of the geometries in the spatial index... will post here once complete.

**Comment by stepps00** [comment]
PIP work for this is still ongoing. I'm hoping substantial progress will be made to hierarchy updates by Monday; I'll follow up here early next week.

**Comment by nvkelso** [comment]
@stepps00 did this finish?

**Comment by stepps00** [comment]
@nkelso unfortunately, no.. still chugging along. I'd estimate the PIP work will be complete by the end of this week.

**Comment by stepps00** [comment]
Another update: I needed to restart the PIP work for this branch yet again, because the process was dreadfully slow. I re-indexed the PIP tools (some instruction [here](https://github.com/whosonfirst/whosonfirst-cookbook/blob/master/how_to/setting_up_pip_service.md)) on Dec 21/22. 

So far, the PIP work on this branch has processed 8,560 of 18,850 records (and descendants), or around 45% of the records. At that pace, I'd estimate this branch will be ready to merge in ~two weeks, with an extra day needed for QA and any remaining small tasks.

I'm also realizing that I merged https://github.com/whosonfirst-data/whosonfirst-data-admin-us/pull/90 in this repo, which introduced ~a dozen merge conflicts to this branch. The conflicts should be relatively simple to resolve (lastmodified dates).. I'll make sure to hold off on additional PRs in this repo until this PR is merged.

**Comment by stepps00** [comment]
Update: 23854 / 31770 rows have been processed through the PIP tools, or roughly 75%. The speed of processing seems to be going slower than estimated in my previous comment.. my hope is this will wrap up in ~3 weeks.

I'll add another comment next week with an update.

**Comment by stepps00** [comment]
Update: Row 28285 / 31770 being processed now. I expect this to be complete in 6-7 days.

**Comment by stepps00** [comment]
Update: the majority of the PIP work for this branch is complete. I'm running a handful of records against the PIP tools for a second time to fix some issues found during QA.

This PR should be ready to merge shortly.

**Comment by stepps00** [comment]
PIP work is complete, merge conflicts have been addressed, and I've QA'd a good amount of records in this branch. This PR is ready for review, over to @nvkelso (again).

File: data/172/943/100/1/1729431001.geojson
Comment (nvkelso) [inline data/172/943/100/1/1729431001.geojson]: What's the rational for this high `min_zoom`? That would never show on the map.

Comment (nvkelso) [reply data/172/943/100/1/1729431001.geojson]: This applies to quite a few of these updates.

Comment (stepps00) [reply data/172/943/100/1/1729431001.geojson]: Ah, that was to address this comment: https://github.com/whosonfirst-data/whosonfirst-data-admin-us/pull/86#issuecomment-726291483

Comment (nvkelso) [reply data/172/943/100/1/1729431001.geojson]: And combined with `"wof:statistical_gore":1`.

File: data/172/939/604/1/1729396041.geojson
Comment (nvkelso) [inline data/172/939/604/1/1729396041.geojson]: For the new `localadmin` we're importing like this one... often times they shouldn't be part of a label hierarchy. But this one is marked true for that... and it's more like a statistical gore than not?

Comment (stepps00) [reply data/172/939/604/1/1729396041.geojson]: Here's the record after all commits on this branch: https://raw.githubusercontent.com/whosonfirst-data/whosonfirst-data-admin-us/stepps00/locality-localadmin/data/172/939/604/1/1729396041.geojson

This record _is_ marked with `"mz:hierarchy_label":1` and does not have a `wof:statistical_gore` property set.

There were 101 new `localadmin` records added to this repo (this record being one of them) - mostly to handle new/removed features from the US Census file. Are we proposing to update properties on all `localadmin`?

Comment (nvkelso) [reply data/172/939/604/1/1729396041.geojson]: I see, let's ignore this for now. 

In Nebraska there are a few of these oddballs that the US Census seems to add in to create a continuous fabric but are otherwise a bit odd. See https://spelunker.whosonfirst.org/id/85688563/descendants/?exclude=nullisland&placetype=localadmin.

**Review by nvkelso — APPROVED** [review]
The mother of all PRs. I spot checked records and reviewed this heads up. Let's pull the trigger and edit forward.

Please address this one todo from original PR description and call it a day:

- Update the `whosonfirst-properties` repo with a better description for `wof:coterminous` and `wof:hierarchy_label`. When both are set on a record, special considerations should be made so users are not appending extra information to an address string.

**Comment by stepps00** [comment]
Opened https://github.com/whosonfirst/whosonfirst-properties/issues/104 to track updates to the coterminous and hierarchy_label properties.

I'm reviewing these changes one last time in QGIS because the changeset is so large.. will merge shortly.

**Pull request merged**
